src/mrdog/change_request/github.rs

```diff
@@ -17,25 +15,63 @@ impl GitHubProvider {
     }
 }
 
+// Test @ https://docs.github.com/en/graphql/overview/explorer
+const QUERY: &str = "
+{
+    viewer {
+      pullRequests(first: 100, states: OPEN) {
+        nodes {
+          id
+          title
+          url
+          author {
+            login
+          }
+          assignees(first: 100) {
+            nodes {
+              login
+            }
+          }
+          reviews(first: 100) {
+            edges {
+              node {
+                id
+              }
+            }
+          }
+          changedFiles
+          additions
+          deletions
+          createdAt
+          reviewDecision
+          headRepository {
+            name
+            url
+          }
+        }
+      }
+    }
+  }
+";
+
 #[async_trait]
 impl ChangeRequestProvider for GitHubProvider {
     async fn fetch(&self) -> Vec<super::ChangeRequest> {
         let instance = Octocrab::builder()
             .personal_token(self.token.clone())
             .build()
             .unwrap();
-        let mrs = instance
-            .search()
-            // TODO: assignee, mentions, review-requested
-            .issues_and_pull_requests("is:open is:pr author:@me archived:false")
-            .send()
-            .await
-            .expect("Request succeeds"); // TODO: Result
-        mrs.into_iter()
+
+        let response: serde_json::Value = instance.graphql(QUERY).await.unwrap();
+        // TODO: Result
+        response["data"]["viewer"]["pullRequests"]["nodes"]
+            .as_array()
+            .unwrap()
+            .iter()
             .map(|i| super::ChangeRequest {
-                id: i.id.to_string(),
-                url: i.url.into(),
-                title: i.title,
+                id: i["id"].as_str().unwrap().to_string(),
```

Comment:
won't this panic is it's `None`? Maybe we could add some default (probably not for id, but url and/or title)

Reply:
Correct, the function formaly did not returned result for sake of simplicity, will update in the future change and maybe add some error handling lib like [anyhow](https://docs.rs/anyhow/latest/anyhow/), good spotting :leopard: